### PR TITLE
[PM-5607] password generator component & configuration

### PR DESCRIPTION
**Pull request by audreyality**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-5607

## 📔 Objective

Implement password generation options component supporting:

* Reactive forms
* component library
* reactive credential generation service

The following aspects from the mockups are out of scope; they'll be included in a wrapper component ([PM-8279](https://bitwarden.atlassian.net/browse/PM-8279)):

* The generated password box
* The password history nav
* Padding around the options

The slider is also out of scope; it's not yet implemented ([PM-7759](https://bitwarden.atlassian.net/browse/PM-7759)).

And there's a known issue with policy not being applied that'll be fixed in [PM-11418](https://bitwarden.atlassian.net/browse/PM-11418). The control properly locks out editing and enforces minimums. The service doesn't fix the settings.

## 📸 Screenshots

image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11418]: https://bitwarden.atlassian.net/browse/PM-11418?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PM-8279]: https://bitwarden.atlassian.net/browse/PM-8279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PM-7759]: https://bitwarden.atlassian.net/browse/PM-7759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser/src/_locales/en/messages.json [hunk 1]
```diff
@@ -383,17 +383,41 @@
...
+    "description": "Label for the password generator numbers checkbox"
+  },
+  "specialCharactersDescription": {
+    "message": "Include special characters",
+    "description": "Full description for the password generator special characters checkbox"
   },
   "specialCharacters": {
-    "message": "Special characters (!@#$%^&*)"
+    "message": "!@#$%^&*",
+    "description": "Label for the password generator special characters checkbox"
```
Comment (djsmith85) [inline apps/browser/src/_locales/en/messages.json:420]: ⚠️  We unfortunately can't change the existing messages, as that would also change them within the existing generator and would not be reflected by existing translation. Please create new keys. We'll remove the old ones during the clean up of the v1 generator

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -414,7 +438,7 @@
   "minSpecial": {
     "message": "Minimum special"
   },
-  "avoidAmbChar": {
+  "ambiguous": {
```
Comment (djsmith85) [inline apps/browser/src/_locales/en/messages.json:441]: ⚠️  This key is also currently used within the old generator. Please create a new key.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/browser/src/_locales/en/messages.json (hunk 1 above)
Comment (audreyality) [reply apps/browser/src/_locales/en/messages.json]: Addressed in c5f2e9cdbd4.

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (audreyality) [reply apps/browser/src/_locales/en/messages.json]: Addressed in c5f2e9cdbd4.

File: apps/browser/src/tools/popup/generator/credential-generator.component.html [hunk 3]
```diff
@@ -1 +1 @@
-<bit-passphrase-settings />
```
[the remark this answers is not in the document]
Comment (audreyality) [reply apps/browser/src/tools/popup/generator/credential-generator.component.html:1]: Yeah. That file will be updated to host whatever the screenshot is for each PR. The penultimate commit will be the full generator component. The last one will just add the history nav underneath.

File: libs/tools/generator/components/src/password-settings.component.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            attr.aria-description="{{ 'specialCharactersDescription' | i18n }}"
+            title="{{ 'specialCharactersDescription' | i18n }}"
+          >
+            <input bitCheckbox type="checkbox" formControlName="special" />
+            <bit-label>{{ "specialCharactersLabel" | i18n }}</bit-label>
+          </bit-form-control>
+        </div>
+        <div class="tw-flex">
+          <bit-form-field class="tw-basis-1/2 tw-mr-4">
+            <bit-label>{{ "minNumbers" | i18n }}</bit-label>
```
Comment (djsmith85) [inline libs/tools/generator/components/src/password-settings.component.html:59]: ⚠️  `minNumbers` has not been defined

File: libs/tools/generator/components/src/password-settings.component.html [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            <input
+              bitInput
+              type="number"
+              [min]="minMinNumber"
+              [max]="maxMinNumber"
+              formControlName="minNumber"
+            />
+          </bit-form-field>
+          <bit-form-field class="tw-basis-1/2">
+            <bit-label>{{ "minSpecial" | i18n }}</bit-label>
```
Comment (djsmith85) [inline libs/tools/generator/components/src/password-settings.component.html:69]: ⚠️  `minSpecial` has not been defined

File: libs/tools/generator/components/src/password-settings.component.html [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            type="number"
+            [min]="minLength"
+            [max]="maxLength"
+          />
+        </bit-form-field>
+      </bit-card>
+    </div>
+    <div>
+      <bit-card>
+        <div class="tw-mb-4">Include</div>
```
Comment (djsmith85) [inline libs/tools/generator/components/src/password-settings.component.html:22]: ⚠️  Missing i18n pipe. `include` is defined in messages.json but is not used here.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/tools/generator/components/src/password-settings.component.html (hunk 4 above)
Comment (audreyality) [reply libs/tools/generator/components/src/password-settings.component.html:59]: [It's already defined](https://github.com/bitwarden/clients/blob/c5f2e9cdbd4e28fda1ec41491d7164e67c6966fe/apps/browser/src/_locales/en/messages.json#L451-L453)

File: libs/tools/generator/components/src/password-settings.component.html (hunk 5 above)
Comment (audreyality) [reply libs/tools/generator/components/src/password-settings.component.html:69]: [It's already defined.])https://github.com/bitwarden/clients/blob/c5f2e9cdbd4e28fda1ec41491d7164e67c6966fe/apps/browser/src/_locales/en/messages.json#L454-L456)

File: libs/tools/generator/components/src/password-settings.component.html (hunk 4 above)
Comment (audreyality) [reply libs/tools/generator/components/src/password-settings.component.html:59]: It's already defined

https://github.com/bitwarden/clients/blob/c5f2e9cdbd4e28fda1ec41491d7164e67c6966fe/apps/browser/src/_locales/en/messages.json#L451-L453

File: libs/tools/generator/components/src/password-settings.component.html (hunk 6 above)
Comment (audreyality) [reply libs/tools/generator/components/src/password-settings.component.html]: I knew when I did this I'd forget to fix it. 🫠 

Addressed in 03d0660ae7f.

**Review by djsmith85 — APPROVED** [review]
This is looking great and ready to get merged.

**Pull request merged**